build/upgrade_pdfjs.sh

```diff
@@ -1,77 +1,74 @@
 echo "-----------------------------------------------------------------------------------"
-echo "Enabling e-signature rendering without validation..."
+echo "Minifying pdf.js files using terser..."
 echo "-----------------------------------------------------------------------------------"
-sed -e 's@;r.setFlags(o.AnnotationFlag.HIDDEN)@@' -i '' ${DOC_STATIC_ASSETS_PATH}/pdf.worker.min.js
+TERSER_ARGS=(--comments false -c sequences=false -m reserved=['__webpack_exports__'] --keep-classnames --keep-fnames --module)
+yarn run terser ${DOC_STATIC_ASSETS_PATH}/pdf.mjs -o ${DOC_STATIC_ASSETS_PATH}/pdf.min.mjs ${TERSER_ARGS[@]}
+yarn run terser ${DOC_STATIC_ASSETS_PATH}/pdf.worker.mjs -o ${DOC_STATIC_ASSETS_PATH}/pdf.worker.min.mjs ${TERSER_ARGS[@]}
+yarn run terser ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.mjs -o ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.min.mjs ${TERSER_ARGS[@]}
 
-# Minify using Google Closure Compiler, options:
-# Output to ES5 (Box supports Chrome, Edge, IE11, Firefox, Safari, and newer versions of iOS, Android)
-# Do not minify pdf.js or pdf.worker.js, as the closure compiler will mangle function names and cause bugs
 echo "-----------------------------------------------------------------------------------"
-echo "Minifying pdf.js files with Google Closure... Warnings are okay!"
+echo "Cleaning up un-minified pdf.js files..."
 echo "-----------------------------------------------------------------------------------"
-java -jar ${DOC_COMPILER_BINARY} --rewrite_polyfills false --language_in ECMASCRIPT_2020 --language_out ECMASCRIPT_2018 --js ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.js --js_output_file ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.min.js
+rm ${DOC_STATIC_ASSETS_PATH}/pdf.mjs
+rm ${DOC_STATIC_ASSETS_PATH}/pdf.worker.mjs
+rm ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.mjs
 
+# we get the cmaps from the prebuild rather than the cdn because we would have to web scrape the cdn in order to get the cmap file names.
+# We don't use the prebuild for the pdf.js files because it doesn't include pdf_viewer
 echo "-----------------------------------------------------------------------------------"
-echo "Minifying pdf.js CSS with cssnano"
+echo "Downloading and copying cmaps to third-party directory..."
 echo "-----------------------------------------------------------------------------------"
-./node_modules/.bin/cssnano ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.css ${DOC_STATIC_ASSETS_PATH}/pdf_viewer.min.css
+curl -L https://github.com/mozilla/pdf.js/releases/download/v${DOC_STATIC_ASSETS_PDFJS_VERSION}/pdfjs-${DOC_STATIC_ASSETS_PDFJS_VERSION}-legacy-dist.zip -o pdfjs-dist.zip
+unzip -q -d pdfjs-dist pdfjs-dist.zip
+\cp -rf pdfjs-dist/web/cmaps ${DOC_STATIC_ASSETS_PATH}/cmaps
+rm -rf pdfjs-dist
+rm pdfjs-dist.zip
 
 echo "-----------------------------------------------------------------------------------"
-echo "Successfully updated and minified pdf.js files!"
+echo "Successfully updated pdf.js files!"
```

Comment:
Do we need to add a reminder to run the  command for the external link fix?

Reply:
Yeah I can add that. The test also gets run as a commit hook